games-strategy/freeciv/freeciv-2.6.0.recipe

```diff
@@ -0,0 +1,128 @@
+Freeciv aims to be mostly rule-compatible with Civilization II [tm], \
+published by Sid Meier and Microprose [tm]. A few rules are different \
+where we think it makes more sense, and we have lots and lots of \
+adjustable parameters to make customizing games possible.
+Freeciv has been implemented completely independently of Civilization; \
+you do not need to own Civilization to play Freeciv."
+HOMEPAGE="http://www.freeciv.org/"
+COPYRIGHT="1996-2018 The Freeciv Team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://netix.dl.sourceforge.net/project/freeciv/Freeciv%202.6/$portVersion/freeciv-$portVersion.tar.bz2"
+CHECKSUM_SHA256="7c20399198d6c7d846fed9a69b02e01134ae5340a3ae0f99d1e38063ade6c999"
+ADDITIONAL_FILES="freeciv.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ -z "${ARCHITECTURES##*!$targetArchitecture*}" ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+GLOBAL_WRITABLE_FILES="
+	settings/freeciv/database.lua keep-old
+	"
+
+PROVIDES="
+	freeciv$secondaryArchSuffix = $portVersion
+	app:FreeCiv = $portVersion
+	cmd:freeciv_manual$commandSuffix
+	cmd:freeciv_sdl$commandSuffix
+	cmd:freeciv_server$commandSuffix
+	lib:libfreeciv$secondaryArchSuffix
+	lib:libfreeciv_srv$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libsdl_1.2$secondaryArchSuffix
+	lib:libsdl_gfx$secondaryArchSuffix
+	lib:libsdl_image_1.2$secondaryArchSuffix
+	lib:libsdl_mixer_1.2$secondaryArchSuffix
+	lib:libsdl_ttf_2.0$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix  >= 5.2
+	devel:liblzma$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libsdl$secondaryArchSuffix
+	devel:libsdl_gfx$secondaryArchSuffix
+	devel:libsdl_image$secondaryArchSuffix
+	devel:libsdl_mixer$secondaryArchSuffix
+	devel:libsdl_ttf$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	autoreconf -vfi
+	CFLAGS="-O2 -pipe" runConfigure --omit-dirs binDir ./configure \
+		--bindir=$commandBinDir --enable-client=sdl --enable-fcmp=no \
+		--disable-sys-lua --enable-shared --disable-static
```

Comment:
Oops! Disabled lua but forgot to remove liblua. Will fix right away :)

Reply:
But the earlier recipes sports lua support? What is your reason for disabling it?

Reply:
Freeciv 2.6 seems to have its own little lua5.3 in its source, so external lua doesn't seem to be needed.

Reply:
It would be much better to use the system provided lua, and AFAIK HaikuPorts guideline prefers the usage of shared system provided libs, instead on the in-tree versions. Please, try to use it if possible.

Reply:
The patch from the previous version might have a fix. I might have to alter the patchset to get it to work with the new version (the build crashes whenever I put the whole patchset in, but I think some parts of the patchset fixes the lua problem). I'll experiment and see if I could get it to run, with lua enabled, and with a patchset 😄

Reply:
Strange. @extrowerk I can't get freeciv to work with lua from the system. It keeps telling me `configure: error: Use of included lua disabled, and no lua found from system` even though I made sure Lua 5.3 and its libraries were installed.

Reply:
@bach5000 : could this be useful? https://github.com/haikuports/haikuports/blob/master/dev-lang/lua/lua-5.3.4.recipe

Reply:
@extrowerk 
Never mind then :) Didn't see it on pkgman.

Reply:
Does freeciv actually use Lua 5.3? The C API for 5.1, 5.2 and 5.3 is different and some projects still require an older version.